### PR TITLE
[importer] Use full abfs path in load data

### PR DESCRIPTION
**Pull request by wing2fly**
## What changes were proposed in this pull request?

When generating SQL for load data, update the abfs path in full path

## How was this patch tested?

add a unit test and manually test in CDW

### CODE REVIEW

File: desktop/libs/indexer/src/indexer/indexers/sql_tests.py [hunk 1]
```diff
@@ -579,3 +582,31 @@ def test_generate_create_empty_kudu_table():
...
 )   STORED AS kudu TBLPROPERTIES("transactional" = "false")
 ;''' in sql, sql)
+
+
+def test_create_ddl_with_abfs():
+  try:
+    from beeswax.server import dbms
+    from azure.conf import ABFS_CLUSTERS
+  except ImportError as e:
+    raise SkipTest('Hive and HiveServer2 interfaces are not enabled')
```
Comment (romainr) [inline desktop/libs/indexer/src/indexer/indexers/sql_tests.py:592]: This should not be needed in latest Hue: beeswax is never blacklisted, as well as azure which is a desktop lib

File: desktop/libs/indexer/src/indexer/indexers/sql_tests.py [hunk 2]
```diff
@@ -579,3 +582,31 @@ def test_generate_create_empty_kudu_table():
...
+
+  finish = ABFS_CLUSTERS.set_for_testing(
+    {
+      'default': {
+        'fs_defaultfs': 'abfs://my-data@yingstorage.dfs.core.windows.net',
```
Comment (romainr) [inline desktop/libs/indexer/src/indexer/indexers/sql_tests.py:597]: Nice test!!

Also could add a very similar one with s3 to check that we don't mess-it up?

File: apps/beeswax/src/beeswax/server/dbms.py [hunk 3]
```diff
@@ -806,7 +807,10 @@ def drop_table(self, database, table):
 
   def load_data(self, database, table, form_data, design, generate_ddl_only=False):
     hql = "LOAD DATA INPATH"
-    hql += " '%(path)s'" % form_data
+    source_path = "%(path)s" % form_data
+    if source_path.lower().startswith("abfs"): #this is to check if its using an ABFS path
```
Comment (romainr) [inline apps/beeswax/src/beeswax/server/dbms.py:811]: Good to know! I don't see a cleaner way for now (until we refactor the importer/file browser).

**Review by romainr — APPROVED** [review]
Nice!

File: desktop/libs/indexer/src/indexer/indexers/sql_tests.py (hunk 2 above)
Comment (wing2fly) [reply desktop/libs/indexer/src/indexer/indexers/sql_tests.py:597]: S3 doesn't have same issue here. Only the abfs URL got @yingstorage.dfs.core.windows.net remove from the path, so Impala or Hive throws InvalidUriAuthorityException for abfs://my-data/test_data/cars.csv.

File: desktop/libs/indexer/src/indexer/indexers/sql_tests.py (hunk 1 above)
Comment (wing2fly) [reply desktop/libs/indexer/src/indexer/indexers/sql_tests.py:592]: Will remove

File: desktop/libs/indexer/src/indexer/indexers/sql_tests.py (hunk 2 above)
Comment (romainr) [reply desktop/libs/indexer/src/indexer/indexers/sql_tests.py:590]: I know, the idea was to make sure we don't mess-up the non abfs path..

**Pull request merged**